### PR TITLE
PSA core 5.12

### PR DESCRIPTION
**Pull request by kfnta**
CC @GuyWi @avolinski @amiraloosh

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 tags
+.DS_Store
```
Comment (AnotherButler) [inline .gitignore:2]: Query: What is this file for? Should it be here?

File: docs/api/security/lifecycle/generate_png.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```
Comment (AnotherButler) [inline docs/api/security/lifecycle/generate_png.sh:1]: Query: What is this file for? Should it be here?

File: docs/api/security/lifecycle/psa_lifecycle.dot [hunk 3]
```diff
@@ -0,0 +1,11 @@
+digraph {
```
Comment (AnotherButler) [inline docs/api/security/lifecycle/psa_lifecycle.dot:1]: Query: What is this file for? Should it be here?

File: docs/api/security/lifecycle/psa_lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+PSA Lifectcle is not a standalone feature as it depends on a PSA bootloader support, which is not yet introduced to mbed-os.
+The only supported lifecycle change available at the moment is `PSA_LIFECYCLE_ASSEMBLY_AND_TEST` to `PSA_LIFECYCLE_ASSEMBLY_AND_TEST`, which can be used for tests to reset device RoT state.
+All the dashed edges are not implemented.
+</span>
+
+Lifecycle can be specified during build time by `MBED_CONF_LIFECYCLE_STATE` macro. Default value is `PSA_LIFECYCLE_ASSEMBLY_AND_TEST`.
+
+In mbed-os PSA Lifecycle is implemented as part of [platform service](../platform_servcie.md)
+
+TODO: link to doxygen
```
Comment (AnotherButler) [inline docs/api/security/lifecycle/psa_lifecycle.md]: Please add an example, in addition to the Doxygen.

Comment (AnotherButler) [reply docs/api/security/lifecycle/psa_lifecycle.md]: Also, what are the use cases for this API?

File: docs/api/security/platform_service.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<h2 id="platform-service">Platfrom Service API</h2>
+
+Platform service introduces system reset and PSA Lifecycle APIs.
+PSA platform service may be extended by target vendors to expose target peripherals only accessible from secure side.
+
+Todo: link to doxygen
```
Comment (AnotherButler) [inline docs/api/security/platform_service.md]: Query: What are the use cases for this API?

Also, please add an example, in addition to the Doxygen.

File: docs/api/security/psa.md [hunk 6]
```diff
@@ -32,11 +32,14 @@ The Mbed implementation of PSA supports the following platform types:
...
-- ARMv8-M: Generation of ARM processors featuring TrustZone-M architecture. PSA support for this platforms is in final stages of development and will be added to the list of platforms supported by the Mbed implementation of PSA shortly.
+- ARMv8-M: Generation of ARM processors featuring TrustZone-M architecture. PSA support for this platforms is based on *specialized* [TrustedFirmware-M](https://www.trustedfirmware.org) implementation.
 
 ### RoT services
 
 The Mbed implementation of PSA provides the following services:
 
 - PSA [RoT](../introduction/glossary.html) internal storage.
 - [PSA Crypto APIs (on GitHub)](https://github.com/ARMmbed/mbed-crypto/tree/development/docs).
+- PSA Attestation - TBD
```
Comment (AnotherButler) [inline docs/api/security/psa.md:43]: I don't like including "TBD" in our docs. Does anyone oppose removing this line until we have an Attestation link?

File: docs/porting/psa/spm.md [hunk 7]
```diff
@@ -6,8 +6,19 @@ For more information about SPM, please refer to [the SPM overview page](../apis/
 
 <span class="notes">This page gives guidelines for silicon partners adding SPM capabilities.</span>
 
+mbed-os
```
Comment (AnotherButler) [inline docs/porting/psa/spm.md:9]: Is this a placeholder for anything, or can we remove this line?

File: docs/porting/psa/spm.md [hunk 8]
```diff
@@ -6,8 +6,19 @@ For more information about SPM, please refer to [the SPM overview page](../apis/
...
 
+#### Platform types
+
+- Non-PSA platform: These are single core ARMv7-M targets. On these targets, the Mbed implementation of PSA provides the same PSA services exposing PSA APIs as it would on PSA targets. The PSA emulation layer allows seamless software portability to more security-oriented targets.
+- Asymmetric Multiprocessing (AMP) systems: Multicore ARMv7-M targets (for example, PSoC6 featuring CM4 and CM0+ cores). On these targets, one of the cores is dedicated to PSA use only and implements SPE.
+- ARMv8-M: Generation of ARM processors featuring TrustZone-M architecture. PSA support for this platforms is based on *specialized* [TrustedFirmware-M](https://www.trustedfirmware.org) implementation.
+
+
+#### JSON target definition
+
```
Comment (AnotherButler) [inline docs/porting/psa/spm.md:21]: Should we add a link to this page: https://os.mbed.com/docs/mbed-os/development/porting/porting-targets.html?

File: docs/porting/psa/spm.md [hunk 9]
```diff
@@ -248,6 +272,12 @@ Processor access    |Secure RAM        |Secure FLASH|Nonsecure RAM      |Nonsecu
 `Secure Write`      |   V              |    V       |        V          |    ?
 `Secure Execute`    |   X?             |    V       |        X          |    ?
 
+
+###  TF-M SPM porting for (ARMv8-M targets)
+
+TF-M HAL functions are defined in `tfm_spm_hal.h`
```
Comment (AnotherButler) [inline docs/porting/psa/spm.md:278]: Let's link to this once the Doxygen merges. Also, is there more to add to this section? It feels sparse.

**Review by AnotherButler — CHANGES_REQUESTED** [review]
Add initial queries and comments before editing.

File: .gitignore (hunk 1 above)
Comment (alzix) [reply .gitignore:2]: these files are annoying artefacts created by MacOS filesystem - https://en.wikipedia.org/wiki/.DS_Store

File: docs/api/security/lifecycle/generate_png.sh (hunk 2 above)
Comment (alzix) [reply docs/api/security/lifecycle/generate_png.sh:1]: used this script to generate the nice PNG state machine graph.
please suggest the best place for it.

File: docs/api/security/lifecycle/psa_lifecycle.dot (hunk 3 above)
Comment (alzix) [reply docs/api/security/lifecycle/psa_lifecycle.dot:1]: used this file in DOT language for describing the lifecycle state machine.
please suggest the best place for it.

File: docs/api/security/lifecycle/psa_lifecycle.md (hunk 4 above)
Comment (alzix) [reply docs/api/security/lifecycle/psa_lifecycle.md]: what kind of example required here?

File: docs/porting/psa/spm.md (hunk 9 above)
Comment (alzix) [reply docs/porting/psa/spm.md:278]: i'm not sure why it is not generated

**Comment by alzix** [comment]
sorry i opened this PR using our team account - closing and will reopen with my account

**Review by GuyWi — COMMENTED** [review]
Please see my review in [PSA core 5.12 #986](https://github.com/ARMmbed/mbed-os-5-docs/pull/986).

**Pull request closed without merging**